kvcache/cache.go

```diff
@@ -0,0 +1,54 @@
+package kvcache
+
+import (
+	"errors"
+
+	"github.com/ollama/ollama/ml"
+)
+
+var (
+	ErrKvCacheFull  = errors.New("could not find a kv cache slot")
+	ErrNotSupported = errors.New("model does not support operation")
+)
+
+type Cache interface {
+	// ** used by model implementations **
+
+	// Sets the active layer of the cache
```

Comment:
I _think_ generally the doc for functions should start with the function name, e.g.

```suggestion
	// SetLayer sets the active layer of the cache
```

Reply:
More: https://go.dev/doc/comment